src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3227,6 +3227,132 @@ tx_gateway_frontend::get_all_transactions() {
       });
 }
 
+ss::future<result<tm_transaction, tx_errc>>
+tx_gateway_frontend::get_tx(kafka::transactional_id tid) {
+    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+
+    if (!shard.has_value()) {
+        vlog(txlog.warn, "can't find a shard for {}", model::tx_manager_ntp);
+        return ss::make_ready_future<result<tm_transaction, tx_errc>>(
+          tx_errc::shard_not_found);
+    }
+
+    return container().invoke_on(
+      *shard,
+      _ssg,
+      [tid](tx_gateway_frontend& self)
+        -> ss::future<result<tm_transaction, tx_errc>> {
+          auto partition = self._partition_manager.local().get(
+            model::tx_manager_ntp);
+          if (!partition) {
+              vlog(
+                txlog.warn,
+                "can't get partition by {} ntp",
```

Comment:
you can be more specific here f.e. `transaction manager partition: {} not found`